apisix/utils/rediscluster.lua

```diff
@@ -0,0 +1,60 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local rediscluster      = require("resty.rediscluster")
+local core              = require("apisix.core")
+local ipairs            = ipairs
+
+local _M = {version = 0.1}
+
+local function new_redis_cluster(conf)
+    local config = {
+        name = conf.redis_cluster_name,
+        serv_list = {},
+        read_timeout = conf.redis_timeout,
+        auth = conf.redis_password,
+        dict_name = "plugin-limit-conn-redis-cluster-slot-lock",
```

Comment:
this function is also being used by `limit-conn` so the dict name needs to be parameterized.

Reply:
OK

Reply:
done